feature/client/src/main/java/com/mifos/feature/client/clientDetails/presentation/ClientDetailsScreen.kt

```diff
@@ -0,0 +1,474 @@
+import com.mifos.core.designsystem.component.MifosSelectImageDialog
+import com.mifos.core.designsystem.component.MifosSweetError
+import com.mifos.core.designsystem.theme.Black
+import com.mifos.core.designsystem.theme.BluePrimary
+import com.mifos.core.designsystem.theme.BluePrimaryDark
+import com.mifos.core.designsystem.theme.White
+import com.mifos.core.objects.accounts.loan.LoanAccount
+import com.mifos.core.objects.accounts.savings.DepositType
+import com.mifos.core.objects.accounts.savings.SavingsAccount
+import com.mifos.core.objects.client.Client
+import com.mifos.feature.client.R
+import java.io.File
+import java.util.Objects
+
+/**
+ * Created by Aditya Gupta on 18/03/24.
+ */
+
+@Composable
+fun ClientDetailsScreen(
+    clientId: Int,
+    onBackPressed: () -> Unit,
+    addLoanAccount: (Int) -> Unit,
+    addSavingsAccount: (Int) -> Unit,
+    charges: (Int) -> Unit,
+    documents: (Int) -> Unit,
+    identifiers: (Int) -> Unit,
+    moreClientInfo: (Int) -> Unit,
+    notes: (Int) -> Unit,
+    pinpointLocation: (Int) -> Unit,
+    survey: (Int) -> Unit,
+    uploadSignature: (Int) -> Unit,
+    loanAccountSelected: (Int) -> Unit,
+    savingsAccountSelected: (Int, DepositType) -> Unit,
+    activateClient: (Int) -> Unit
+) {
+
+    val clientDetailsViewModel: ClientDetailsViewModel = hiltViewModel()
+    val context = LocalContext.current
+    val state = clientDetailsViewModel.clientDetailsUiState.collectAsState().value
+    val client = rememberSaveable { mutableStateOf<Client?>(null) }
+    val loanAccounts = rememberSaveable { mutableStateOf<List<LoanAccount>?>(null) }
+    val savingsAccounts = rememberSaveable { mutableStateOf<List<SavingsAccount>?>(null) }
+    var clientNotFoundError by rememberSaveable { mutableStateOf(false) }
+
+    var showMenu by remember { mutableStateOf(false) }
+    val showSelectImageDialog = remember { mutableStateOf(false) }
+    var imageUri by remember { mutableStateOf<Uri?>(null) }
+
+    val snackbarHostState = remember { SnackbarHostState() }
+
+    val file = context.createImageFile()
+    val cameraImageUri = FileProvider.getUriForFile(
+        Objects.requireNonNull(context),
+        "com.mifos.mifosxdroid" + ".provider",
+        file
+    )
+    val permissionState = rememberMultiplePermissionsState(
+        permissions = listOf(
+            Manifest.permission.CAMERA
+        )
+    )
+
+    val galleryLauncher = rememberLauncherForActivityResult(
+        contract = ActivityResultContracts.GetContent(),
+        onResult = { uri ->
+            uri?.let {
+                imageUri = it
+                val bitmap = context.contentResolver.openInputStream(uri).use { stream ->
+                    BitmapFactory.decodeStream(stream).asImageBitmap().asAndroidBitmap()
+                }
+                showSelectImageDialog.value = false
+                clientDetailsViewModel.saveClientImage(clientId, bitmap)
```

Comment:
<img width="613" alt="image" src="https://github.com/openMF/android-client/assets/5724482/1e99fb33-18de-4a87-8515-b0adecf5c890">
This can be converted into function and used here and in cameraLauncher

Reply:
it will create more complexity and if we do then we need to pass as mutablestateof type ,context, and viewmodel in function.